### ISSUE TITLE
requires android.permission.BLUETOOTH_CONNECT

### ISSUE THREAD

**Issue by Nishit-Chauhan** [yashtops/react-native-cubetap#1]
@yashtops I Have added required permission to AndroidManifest.xml file and try to use sample code and getting following error

```
import React, { useEffect, useState } from 'react'
import { View, Text } from 'react-native'
import {
  isBluetoothEnabled,
  requestPermissions,
  enableBluetooth,
  startDiscovery,
  getCubetapeDevice
} from 'react-native-cubetapsdk'

const Bluetooth = () => {

  const [permissionStatus, setPermissionStatus] = useState(false)
  const [bluetoothEnabled, setBluetoothEnabled] = useState(false)

  useEffect(() => {
    checkPermission()
  }, [])

  const checkPermission = async () => {
    const permissionStatus = await requestPermissions()
  }

  useEffect(() => {
    if (permissionStatus) {
      checkBluetoothStatus()
    }
  }, [permissionStatus])

  useEffect(() => {
    if (bluetoothEnabled) {
      getBluetoothDevice()
    } else {
      enableBluetooth()
    }
  }, [bluetoothEnabled])

  const checkBluetoothStatus = async () => {
    const check = await isBluetoothEnabled()
    setBluetoothEnabled(check)
  }

  const getBluetoothDevice = async () => {
    const device = await getCubetapeDevice();
  }


  return (
    <View>
      <Text>BluetoothSDK</Text>
    </View>
  )
}

export default Bluetooth

```
![1716524196759](https://github.com/yashtops/react-native-cubetap/assets/45855322/197ef868-471a-4a51-9a6e-a6c83d466e70)

![Screenshot 2024-05-24 at 9 50 58 AM](https://github.com/yashtops/react-native-cubetap/assets/45855322/fc9368fc-c39f-4a7c-8677-37ddc4aca053)

**Comment by yashtops** [comment]
Hello @Nishit-Chauhan 
There was a issue with the import in the usage example.i have update readme
I have publish the new version [v1.0.3](https://github.com/yashtops/react-native-cubetap/releases/tag/beta1.0.3)
please try and let me know

**Comment by Nishit-Chauhan** [comment]
Thanks @yashtops 
I Have installed latest version and facing the module throw error before user interact with permission popup

also if location service is not enabled then I think popup should ask user to enable location service first

![1716536589377](https://github.com/yashtops/react-native-cubetap/assets/45855322/08c6605e-a348-4e3a-9f1f-0f1471891765)
![1716536585674](https://github.com/yashtops/react-native-cubetap/assets/45855322/b18a0ad7-c174-4ea9-9504-b8814ac89653)

**Comment by yashtops** [comment]
Fixed in [v1.0.4](https://github.com/yashtops/react-native-cubetap)
